src/config/schema.rs

```diff
@@ -126,10 +133,52 @@ pub struct S3 {
     pub cache_properties: Option<ObjectCacheProperties>,
 }
 
+impl S3 {
+    pub fn from_bucket_and_options(
+        bucket: String,
+        map: &HashMap<String, String>,
+    ) -> Result<Self, DataFusionError> {
+        Ok(S3 {
+            region: map.get("region").cloned(),
+            access_key_id: map
+                .get("access_key_id")
+                .ok_or(DataFusionError::Execution(
+                    "'access_key_id' not found in provided options".to_string(),
+                ))?
+                .clone(),
+            secret_access_key: map
+                .get("secret_access_key")
+                .ok_or(DataFusionError::Execution(
+                    "'secret_access_key' not found in provided options".to_string(),
+                ))?
```

Comment:
`DataFusionError` feels a bit foreign inside the config module.

Reply:
Yeah, we could define something ourselves or reuse a serde parsing error, given this is part of the config deserialization logic?

Reply:
Right; tbh I'm still torn torn between doing this here (or somewhere else in our code), or going with the datafusion-cli helper functions (which would make this redundant).

Reply:
Looks like it's 61MB regardless we include `datafusion-cli` or not. I was also wary of using the df-cli functions as they initially source values from the env, and delta-rs used to dump options values to env vars when creating a new external table, but it seems this is no longer the case.

Given this, I eill try this approach next.

Reply:
Turns out delta-rs still dumps the env vars when creating an external delta table on S3, which would interfere with the creation of a new object store in [get_s3_object_store_builder](https://docs.rs/datafusion-cli/latest/datafusion_cli/object_storage/fn.get_s3_object_store_builder.html).

So I'll leave the implementation as is for now (only changing the error above and adding some tests).